### DISCUSSION TITLE
XRDP remote timeout settings

### DISCUSSION THREAD

**Discussion by Harsh-ECX** [neutrinolabs/xrdp#2795] [Q&A] [answered]
I want to change my XRDP timeout settings: -

MY idle session disconnected after every 2mins; I have changed required configuration file at /etc/xrdp/sesman.ini

***************************************************

[Sessions]
;; X11DisplayOffset - x11 display number offset
; Type: integer
; Default: 10
X11DisplayOffset=10

;; MaxSessions - maximum number of connections to an xrdp server
; Type: integer
; Default: 0
MaxSessions=5

;; IdleTimeLimit - when to disconnect idle sessions
; Type: integer
; Default: 0
; If set to 0, automatic disconnection is disabled.
; if not zero, the seconds before a disconnected session is killed
IdleTimeLimit=300

;; KillDisconnected - kill disconnected sessions
; Type: boolean
; Default: false
; if 1, true, or yes, kill session after 60 seconds
KillDisconnected=true

;; DisconnectedTimeLimit - when to kill idle sessions
; Type: integer
; Default: 0
; if not zero, the seconds before a disconnected session is killed
; min 60 seconds
DisconnectedTimeLimit=300

***************************************************

Can anyone help me out with this issue! 
Thanks!

**Comment by matt335672** [comment]
@Harsh-ECX - a couple of questions for you:-
1) xrdp version and operating system?
2) Are you using the xorgxrdp back-end?
3) The above settings look like they should cause a timeout of 5 minutes. Are you saying that you are getting 2 minutes instead?

Thanks.

**Reply by Harsh-ECX** [reply]
@matt335672 --> Thanks for looking into this issue,
Let me answer your asked question below: -
1. Xrdp version --> Version: 0.9.5-2, operating system --->5.4.0-1109-azure #115~18.04.1-Ubuntu

2. Yes, dpkg --list |grep xrdp
ii  xorgxrdp                     0.9.5-2                                 amd64        Remote Desktop Protocol (RDP) modules for X.org
ii  xrdp                            0.9.5-2                                 amd64        Remote Desktop Protocol (RDP) server

3.  Yes, I have modified the configuration file for 5min timeout for idle sessions, but the session disconnects after 2mins idle. 
Let me know if you require any other details for the same.

**Reply by matt335672** [reply, marked as answer]
There's only a limited amount we can do with that version - it's pretty old. Ubuntu 18.04 went out of support at the end of May, and I can't find any information on what exactly was packaged with your xorgxrdp version - we don't ship a v0.9.5-2 of xorgxrdp so I'm guessing it's what we shipped as 0.2.5.

Your connection is either terminating because the 5 minutes isn't being honoured, or because something else is terminating it. Can you remember if it was terminating before you modified the config?

**Reply by Harsh-ECX** [reply]
there were some other configurations that was causing the termination of my xrdp session, thank you for your help and support.

**Reply by vdrover** [reply]
> there were some other configurations that was causing the termination of my xrdp session, thank you for your help and support.

Can you list them and how you solved it?

**Reply by Harsh-ECX** [reply]
@vdrover try to change your /etc/xrdp/sesman.ini as mine maybe it will help

;; See man 5 sesman.ini for details

[Globals]
ListenAddress=127.0.0.1
ListenPort=3350
EnableUserWindowManager=true
; Give in relative path to user's home directory
UserWindowManager=startwm.sh
; Give in full path or relative path to /etc/xrdp
DefaultWindowManager=startwm.sh
; Give in full path or relative path to /etc/xrdp
ReconnectScript=reconnectwm.sh

[Security]
AllowRootLogin=true
MaxLoginRetry=4
TerminalServerUsers=tsusers
TerminalServerAdmins=tsadmins
; When AlwaysGroupCheck=false access will be permitted
; if the group TerminalServerUsers is not defined.
AlwaysGroupCheck=false
; When RestrictOutboundClipboard=true clipboard from the
; server is not pushed to the client.
RestrictOutboundClipboard=false

[Sessions]
;; X11DisplayOffset - x11 display number offset
; Type: integer
; Default: 10
X11DisplayOffset=10

;; MaxSessions - maximum number of connections to an xrdp server
; Type: integer
; Default: 0
MaxSessions=50

;; KillDisconnected - kill disconnected sessions
; Type: boolean
; Default: false
; if 1, true, or yes, every session will be killed within DisconnectedTimeLimit
; seconds after the user disconnects
KillDisconnected=true

;; DisconnectedTimeLimit (seconds) - wait before kill disconnected sessions
; Type: integer
; Default: 0
; if KillDisconnected is set to false, this value is ignored
DisconnectedTimeLimit=0

;; IdleTimeLimit (seconds) - wait before disconnect idle sessions
; Type: integer
; Default: 0
; Set to 0 to disable idle disconnection.
IdleTimeLimit=120

;; Policy - session allocation policy
; Type: enum [ "Default" | "UBD" | "UBI" | "UBC" | "UBDI" | "UBDC" ]
; "Default" session per <User,BitPerPixel>
; "UBD" session per <User,BitPerPixel,DisplaySize>
; "UBI" session per <User,BitPerPixel,IPAddr>
; "UBC" session per <User,BitPerPixel,Connection>
; "UBDI" session per <User,BitPerPixel,DisplaySize,IPAddr>
; "UBDC" session per <User,BitPerPixel,DisplaySize,Connection>
Policy=Default

[Logging]
; Note: Log levels can be any of: core, error, warning, info, debug, or trace
LogFile=xrdp-sesman.log
LogLevel=INFO
EnableSyslog=true
#SyslogLevel=INFO
#EnableConsole=false
#ConsoleLevel=INFO
#EnableProcessId=false

[LoggingPerLogger]
; Note: per logger configuration is only used if xrdp is built with
; --enable-devel-logging
#sesman.c=INFO
#main()=INFO

;
; Session definitions - startup command-line parameters for each session type
;

[Xorg]
; Specify the path of non-suid Xorg executable. It might differ depending
; on your distribution and version. Find out the appropreate path for your
; environment. The typical path is known as follows:
;
; Fedora 26 or later : param=/usr/libexec/Xorg
; Debian 9 or later : param=/usr/lib/xorg/Xorg
; Ubuntu 16.04 or later : param=/usr/lib/xorg/Xorg
; Arch Linux : param=/usr/lib/Xorg
; CentOS 7 : param=/usr/bin/Xorg or param=Xorg
; CentOS 8 : param=/usr/libexec/Xorg
;
param=/usr/lib/xorg/Xorg
; Leave the rest paramaters as-is unless you understand what will happen.
param=-config
param=xrdp/xorg.conf
param=-noreset
param=-nolisten
param=tcp
param=-logfile
param=.xorgxrdp.%s.log

[Xvnc]
param=Xvnc
param=-bs
param=-nolisten
param=tcp
param=-localhost
param=-dpi
param=96

[Chansrv]
; drive redirection
; See sesman.ini(5) for the format of this parameter
#FuseMountName=/run/user/%u/thinclient_drives
#FuseMountName=/media/thinclient_drives/%U/thinclient_drives
FuseMountName=thinclient_drives
; this value allows only the user to acess their own mapped drives.
; Make this more permissive (e.g. 022) if required.
FileUmask=077
; Can be used to disable FUSE functionality - see sesman.ini(5)
#EnableFuseMount=false

[ChansrvLogging]
; Note: one log file is created per display and the LogFile config value
; is ignored. The channel server log file names follow the naming convention:
; xrdp-chansrv.${DISPLAY}.log
;
; Note: Log levels can be any of: core, error, warning, info, debug, or trace
LogLevel=INFO
EnableSyslog=true
#SyslogLevel=INFO
#EnableConsole=false
#ConsoleLevel=INFO
#EnableProcessId=false

[ChansrvLoggingPerLogger]
; Note: per logger configuration is only used if xrdp is built with
; --enable-devel-logging
#chansrv.c=INFO
#main()=INFO

[SessionVariables]
PULSE_SCRIPT=/etc/xrdp/pulse/default.pa
